src/find/matchers/type_matcher.rs

```diff
@@ -0,0 +1,75 @@
+use std::error::Error;
+use std::fs::FileType;
+use walkdir::DirEntry;
+
+use find::matchers::{Matcher, MatcherIO};
+
+/// This matcher checks the type of the file.
+pub struct TypeMatcher {
+    file_type_fn: fn(&FileType) -> bool,
+}
+
+impl TypeMatcher {
+    pub fn new(type_string: &str) -> Result<TypeMatcher, Box<Error>> {
```

Comment:
I prefer the consistency of every Matcher having a new method (and a new_box), rather than ones that take a single string using FromStr and all the others using new.

There's a difference between "this is an object that can be represented as a string" and "this is an object that happens to take a single string as an argument".